File: Ahorn/triggers/teleportTrigger.jl
module DJMapHelperTeleportTrigger

using ..Ahorn, Maple

@mapdef Trigger "DJMapHelper/teleportTrigger" Teleport(x::Integer, y::Integer, width::Integer=16, height::Integer=16, bonfire::Bool=false, room::String="", spawnPointX::Int=0, spawnPointY::Int=0, Dreaming::String="NoChange", KeepKey::Bool=true, introTypes::String="WakeUp")

const dreams = String[
    "Dreaming",
    "Awake",
    "NoChange"
]

const introTypesList = String[
    "Respawn",
    "WalkInRight",
    "WalkInLeft",
    "Jump",
    "WakeUp",
    "Fall",
    "TempleMirrorVoid",
    "None",
    "ThinkForABit"
]

const placements = Ahorn.PlacementDict(
    "Teleport Trigger (DJMapHelper)" => Ahorn.EntityPlacement(
        Teleport,
        "rectangle"
    )
)

Ahorn.editingOptions(Trigger::Teleport) = Dict{String, Any}(
    "Dreaming" => dreams,
    "introTypes" => introTypesList
)

end

File: Loenn/triggers/teleport_trigger.lua
local teleportTrigger = {}

teleportTrigger.name = "DJMapHelper/teleportTrigger"
teleportTrigger.fieldOrder = {
    "x", "y", "width", "height", "Dreaming", "bonfire", "KeepKey", "introTypes", "room", "spawnPointX", "spawnPointY"
}
teleportTrigger.placements = {
    name = "normal",
    data = {
        bonfire = true,
        KeepKey = true,
        Dreaming = "NoChange",
        introTypes = "WakeUp",
        room = "",
        spawnPointX = 0,
        spawnPointY = 0,
    }
}
teleportTrigger.fieldInformation = {
    spawnPointX = { fieldType = "integer" },
    spawnPointY = { fieldType = "integer" },
    Dreaming = {
        fieldType = "anything",
        options = { "Dreaming", "Awake", "NoChange" },
        editable = false
    },
    introTypes = {
        fieldType = "anything",
        options = {
            "Respawn",
            "WalkInRight",
            "WalkInLeft",
            "Jump",
            "WakeUp",
            "Fall",
            "TempleMirrorVoid",
            "None",
            "ThinkForABit"
        },
        editable = false
    }
}

return teleportTrigger